release-notes/7.0/preview/api-diff/preview1/Microsoft.NETCore.App/7.0-preview1_System.Security.Cryptography.md

```diff
@@ -0,0 +1,140 @@
+# System.Security.Cryptography
+
+``` diff
+ namespace System.Security.Cryptography {
+     public abstract class AsymmetricAlgorithm : IDisposable {
++        public string ExportEncryptedPkcs8PrivateKeyPem(ReadOnlySpan<char> password, PbeParameters pbeParameters);
++        public string ExportPkcs8PrivateKeyPem();
++        public string ExportSubjectPublicKeyInfoPem();
++        public bool TryExportEncryptedPkcs8PrivateKeyPem(ReadOnlySpan<char> password, PbeParameters pbeParameters, Span<char> destination, out int charsWritten);
++        public bool TryExportPkcs8PrivateKeyPem(Span<char> destination, out int charsWritten);
++        public bool TryExportSubjectPublicKeyInfoPem(Span<char> destination, out int charsWritten);
+     }
++    public abstract class ECAlgorithm : AsymmetricAlgorithm
```

Comment:
@carlossanlop, @bartonjs, what happened here?  At a minimum the diff is messed up as there's no definition shown for ECAlgorithm.

Reply:
@safern helped me figure it out. We missed an argument when running the tool. It will now show up complete in the next commit I'll push:

```diff
+    public abstract class ECAlgorithm : AsymmetricAlgorithm {
+        protected ECAlgorithm();
+        public virtual byte[] ExportECPrivateKey();
+        public string ExportECPrivateKeyPem();
+        public virtual ECParameters ExportExplicitParameters(bool includePrivateParameters);
+        public virtual ECParameters ExportParameters(bool includePrivateParameters);
+        public virtual void GenerateKey(ECCurve curve);
+        public virtual void ImportECPrivateKey(ReadOnlySpan<byte> source, out int bytesRead);
+        public override void ImportEncryptedPkcs8PrivateKey(ReadOnlySpan<byte> passwordBytes, ReadOnlySpan<byte> source, out int bytesRead);
+        public override void ImportEncryptedPkcs8PrivateKey(ReadOnlySpan<char> password, ReadOnlySpan<byte> source, out int bytesRead);
+        public override void ImportFromEncryptedPem(ReadOnlySpan<char> input, ReadOnlySpan<byte> passwordBytes);
+        public override void ImportFromEncryptedPem(ReadOnlySpan<char> input, ReadOnlySpan<char> password);
+        public override void ImportFromPem(ReadOnlySpan<char> input);
+        public virtual void ImportParameters(ECParameters parameters);
+        public override void ImportPkcs8PrivateKey(ReadOnlySpan<byte> source, out int bytesRead);
+        public override void ImportSubjectPublicKeyInfo(ReadOnlySpan<byte> source, out int bytesRead);
+        public virtual bool TryExportECPrivateKey(Span<byte> destination, out int bytesWritten);
+        public bool TryExportECPrivateKeyPem(Span<char> destination, out int charsWritten);
+        public override bool TryExportEncryptedPkcs8PrivateKey(ReadOnlySpan<byte> passwordBytes, PbeParameters pbeParameters, Span<byte> destination, out int bytesWritten);
+        public override bool TryExportEncryptedPkcs8PrivateKey(ReadOnlySpan<char> password, PbeParameters pbeParameters, Span<byte> destination, out int bytesWritten);
+        public override bool TryExportPkcs8PrivateKey(Span<byte> destination, out int bytesWritten);
+        public override bool TryExportSubjectPublicKeyInfo(Span<byte> destination, out int bytesWritten);
+    }
```